sys/kern/cred_syscalls.c

```diff
@@ -0,0 +1,87 @@
+#include <sys/cred.h>
+#include <sys/proc.h>
+#include <sys/errno.h>
+
+int do_getresuid(proc_t *p, uid_t *ruid, uid_t *euid, uid_t *suid) {
+  cred_t cr = p->p_cred;
+  if (ruid != NULL)
```

Comment:
It'd be more effective to use a pointer to `cred_t` instead of creating a local copy of the structure on the stack.
```
cred_t *cr = &p->p_cred;
if (ruid != NULL)
  *ruid = cr->cr_ruid;
[...]
```